middleware.ts

```diff
@@ -56,6 +56,10 @@ export async function middleware(req: NextRequest) {
     return res;
   }
 
+  if (/^\/workspaces\/[^[\/]+$/.test(req.nextUrl.pathname)) {
```

Comment:
this will only match `/workspaces/some-id` for a path. If the `$` is omitted, you get an infinite redirect from a valid workspaces URL.